### ISSUE TITLE
calcite-input-text doesn't show anything when pattern is not matched

### ISSUE THREAD

**Issue by MikeTschudi** [Esri/calcite-design-system#11148]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

The sample line
```html
 <calcite-input-text id="srcId" max-length="32" min-length="32" pattern="[a-f0-9A-F]{32}" placeholder="Item, group, or folder id"></calcite-input-text>
```
only honors the max-length restriction. There's no lack-of-validity display when the text is too short or doesn't match the pattern.  (Also tried with "^" prefix and "$" suffix.)

### Expected Behavior

Highlights like we see in MDN, e.g., if value is too short:
![image](https://github.com/user-attachments/assets/36f0443d-3cab-4371-8bf3-e628b31f04c3)

MDN playground source:
```html
  <label>
    Enter AGO id <input
      pattern="[a-f0-9A-F]{32}"
      size="32"
    />
  </label>
```

### Reproduction Sample

https://codepen.io/miketschudi/pen/GgKEaPY?editors=100

### Reproduction Steps

above CodePen is for 3.0.0-next.81.  For 2.13.2, 
https://codepen.io/miketschudi/pen/WbeOBLV?editors=100

### Reproduction Version

3.0.0-next.81

### Relevant Info

Windows 11 Enterprise using Chrome 131.0

Here's what the component shows when given "not an id!" in the calcite-input-text field:
![image](https://github.com/user-attachments/assets/67950b1b-324c-43c0-8564-de045ec6e84d)

### Priority impact

impact - p3 - not time sensitive

### Impact

no effect yet--just found this in a prototype for future work

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Solutions

**Comment by benelan** [comment]
The current validation logic only works within forms. I updated the provided codepen and it appears to be working correctly: form submission is prevented when you press Enter while the input is focused.

https://codepen.io/benizzy/pen/YPKYxRM?editors=1000

**Comment by benelan** [comment]
Yeah that's a good idea; thanks for the suggestion. cc @DitwanP @geospatialem

We hope to remove that requirement when we migrate to elementInternals in #8126.

**Comment by MikeTschudi** [comment]
@geospatialem, I think that mentioning under [pattern](https://developers.arcgis.com/calcite-design-system/components/input-text/#api-reference-properties-pattern) that it requires a form would help a lot, along with a link to the general info page about [patterns](https://developers.arcgis.com/calcite-design-system/core-concepts/#patterns) that you mentioned.

The latter could be amended to clarify that patterns don't do anything outside of a form. The existing writeup presents patterns as a helpful tool for form validation without mentioning the dependency on forms. So a link from the `input-text` `pattern` to the forms page alone wouldn't be clear.

**Comment by DitwanP** [comment]
🍡✍️ Verified locally
